reader.js

```diff
@@ -7,15 +9,156 @@ const CLOSE_TIMEOUT = 10000;
+                    query += ` where mission = '${mission}'`;
+
+                    influx.query(query).then(result => {
+
+                        let ids = {};
+                        result.forEach(measure => {
+                            if (!ids.hasOwnProperty(measure.id)) {
+                                ids[measure.id] = measure.time;
+                            }
+                        });
+
+                        response.setHeader("Access-Control-Allow-Origin", '*');
+                        response.end(JSON.stringify(ids));
+
+                    }).catch(function (err) {
+                        console.error(`Error querying data from InfluxDB! ${err.stack}`);
+
+                        if (WSS.readyState !== WebSocket.OPEN) {
+                            return;
+                        }
+
+                        WSS.send(id + ':error:' + e);
+                    });
+                }).catch(function (err) {
+                    console.error(`Error getting measurements! ${err.stack}`);
+
+                    if (WSS.readyState !== WebSocket.OPEN) {
+                        return;
+                    }
+
+                    WSS.send(id + ':error:' + e);
+                });
+            }).catch(function (err) {
+                console.error(`Error connecting to InfluxDB! ${err.stack}`);
+
+                if (WSS.readyState !== WebSocket.OPEN) {
+                    return;
+                }
+
+                WSS.send(id + ':error:' + e);
+            });
+
+        } else {
+
+            let mission = requestQuery.pathname.substring(1);
+
+            if (isNaN(mission)) {
+                return;
+            }
+
+            let ids = requestQuery.searchParams.getAll('ids[]');
+
+            influxConnection.then(function (influx) {
+
+                influx.getMeasurements().then(names => {
+
+                    let queries = [];
+                    ids.forEach(id => {
+
+                        let query = `select * from`;
+                        names.forEach(name => {
+                            query += ` ${name},`;
+                        });
+                        query = query.slice(0, -1);
+                        query += ` where (mission = '${mission}') and (id = '${id}')`;
+                        queries.push(query);
+
+                    });
+
+                    influx.query(queries).then(result => {
+
+                        let transmissions = [];
+                        result.forEach(point => {
+                            let transmission = {};
+                            let names = point.groupRows.map((groupRow) => groupRow.name);
+                            point.forEach(measurement => {
+                                transmission[names.shift()] = measurement.value;
+                            });
+                            transmissions.push(transmission);
+                        });
+
+                        response.setHeader("Access-Control-Allow-Origin", '*');
```

Comment:
We could move this up to the top of the function, since we always want to do this.